main.cpp

```diff
@@ -402,3 +403,157 @@ int main(int argc, char** argv) {
 
     return 0;
 }
+else {
+    //User wants to Restore to Stock
+    //Check if Backup Exists
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (wave.rpx)");
+    PresentMenu();
+    if (!wave_state.backup_exists) {
+        //No Backup
+    printf("Failed - No Backup Exists\n");
+            while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_NO_BACKUP); PresentMenu(); }
+            return -1;
+    }
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (nn_olv.rpl)");
+    PresentMenu();
+     std::ifstream is(nn_olv_path, std::ios::binary);
+        auto hash = rpx_hash(is);
+    if (hash.id == CURRENT_PRETENDO_NN_OLV || !hash.patch) {
+        //No Backup
+    printf("Failed - No Backup Exists\n");
+            while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_NO_BACKUP); PresentMenu(); }
+            return -1;
+    }
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (NSSL cert)");
+    PresentMenu();
+    std::ifstream is2(cert_bak_path, std::ios::binary);
+     auto hash2 = cert_hash(is2);
+     if (hash2.patch != CERT_PATCH_STATE_STOCK) {
+        //No Backup
+    printf("Failed - No Backup Exists\n");
+            while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_NO_BACKUP); PresentMenu(); }
+            return -1;
+    }
+
+    //Restore Backup
+RenderMenuLoading(task_percent(task++), "Restoring Miiverse backup... (wave.rpx)");
+    PresentMenu();
+
+    bret = backup_rpx(B_RESTORE_BACKUP, wave_path, wave_bak_path);
+    if (!bret) {
+        printf("Failed to restore wave!\n");
+        wave_strategy.patch_action = P_DO_NOTHING;
+        while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_RESTORE_FAIL); PresentMenu(); }
+        return -1;
+    }
+    RenderMenuLoading(task_percent(task++), "Restoring Miiverse backup... (nn_olv.rpl)");
+    PresentMenu();
+
+    bret = backup_rpx(B_RESTORE_BACKUP, pn_olv_path, nn_olv_path);
+    if (!bret) {
+        printf("Failed to restore olv!\n");
+        olv_strategy.patch_action = P_DO_NOTHING;
+        while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_RESTORE_FAIL); PresentMenu(); }
+        return -1;
+    }
+    RenderMenuLoading(task_percent(task++), "Restoring Miiverse backup... (NSSL cert)");
+    PresentMenu();
+
+    bret = backup_cert(B_RESTORE_BACKUP, cert_path, cert_bak_path);
+    if (!bret) {
+        printf("Failed to restore cert!\n");
+        cert_strategy.patch_action = P_DO_NOTHING;
+        while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_BACKUP_FAIL); PresentMenu(); }
+        return -1;
+    }
+ RenderMenuLoading(task_percent(task++), "Verifying final patches... (wave.rpx)");
+    PresentMenu();
+
+    {
+        std::ifstream is(wave_path, std::ios::binary);
+        auto hash = rpx_hash(is);
+        if (hash.patch != RPX_PATCH_STATE_STOCK) {
+            printf("Failed to commit patches - stock wave in place\n");
```

Comment:
This isn't true - the check just showed the rpx is *not* stock.

Reply:
Whoops forgot to change that